README.md

```diff
@@ -60,6 +60,7 @@ npm test        # run tests
 ### References
 * http://www.inf.fh-flensburg.de/lang/algorithmen/pattern/kmpen.htm
 * https://people.ok.ubc.ca/ylucet/DS/KnuthMorrisPratt.html
+* https://www.scaler.com/topics/data-structures/
```

Comment:
It would be appropriate to give the direct link to the topic: https://www.scaler.com/topics/data-structures/kmp-algorithm/
But regardless, I don't feel comfortable linking to a for-profit corporation site.